### DISCUSSION TITLE
Switching CNI Issue: Replacing Canal with Weave Net

### DISCUSSION THREAD

**Discussion by KaylCoulston** [rancher/rke2#2247] [Q&A] [answered]
I am running a air-gap install with Canal disabled upon installation and Weave Net automatically installed via a manifest in rke2's manifest directory (/var/lib/rancher/rke2/server/manifests/). Weave Net runs but rke2-coredns fails to start along with other pods due to rke2-coredns failing. The rke2-coredns pod and other non kube-system pods show no endpoints along with rejection rules in the iptables list. The rke2-coredns pod also has repeating `no route to host` errors after it is started. 

**Questions**
- Is there anything you need to do when running Weave Net rather than Canal? 
- What could be causing rke2-coredns to have `no route to host` errors?
- Does anyone know why rejection rules for endpoints that are within weave-net's [default range](https://www.weave.works/docs/net/latest/tasks/ipam/configuring-weave/) are being added to the rejection list?

**What I have done**
- Added [loopback ](https://github.com/containernetworking/plugins/releases) (along with other plugins) to the /opt/cni/bin directory.
- firewalld is disabled 
- Running Weave's suggested ip commands: `ip route`, `ip -4 -o addr`, `sudo iptables-save`
- In the case the startup order mattered, I disabled rke-coredns upon installation and once Weave Net was fully started up I started rke2-cordns using `kubectl apply`. The same errors persisted as before.
- Tested Weave Net and Multus. Everything seems to be working as expected. 

**Environment**
- RKE2: 1.20.7
- Weave Net: 2.7.0
- Iptables 1.4.21
- CentOS 7.9

**rke2-coredns logs**
> E1210 01:19:31.189458       1 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.4/tools/cache/reflector.go:105: Failed to list *v1.Service: Get "https://10.43.0.1:443/api/v1/services?limit=500&resourceVersion=0": dial tcp 10.43.0.1:443: connect: no route to host
E1210 01:19:31.189926       1 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.4/tools/cache/reflector.go:105: Failed to list *v1.Endpoints: Get "https://10.43.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0": dial tcp 10.43.0.1:443: connect: no route to host
E1210 01:19:31.190022       1 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.4/tools/cache/reflector.go:105: Failed to list *v1.Namespace: Get "https://10.43.0.1:443/api/v1/namespaces?limit=500&resourceVersion=0": dial tcp 10.43.0.1:443: connect: no route to host
.:53

**Comment by KaylCoulston** [comment, marked as answer]
Setting Weave Nets ip range to RKE2's ip range solved the issue.

**Reply by fairmonk** [reply]
How did you run air-gap installation with canal disabled ?
I need to air-gap install rke2 with calico, if you can provide a hint I kindly appreciate it ( only managed to install with default canal cni plugin, using tarball images )

**Reply by KaylCoulston** [reply]
You should be able to just [configure ](https://docs.rke2.io/install/network_options) the cluster to use calico. Weave isn't supported by rancher anymore so we used ranchers bootstrap method to load the image and deploy the manifest. Both [RKE2](https://docs.rke2.io/advanced#auto-deploying-manifests) and [K3S](https://docs.k3s.io/advanced#auto-deploying-manifests) support deploying manifests. Then we put the image in /var/lib/rancher/rke2/agent/images.
